gulpfile.js

```diff
@@ -198,10 +198,10 @@ async function updateBuildNumber(args) {
             versionParts.length > 2 ? versionParts[2].replace(/(\d+)/, args.buildNumber) : args.buildNumber;
         const newVersion =
             versionParts.length > 1
-                ? `${versionParts[0]}.${versionParts[1]}.${versionParts[0].substring(
```

Comment:
This was the problem we were using [0].
Also, using the last n characters is incorrect, we should use the first n characters, ignoring the last 3 characters.

For instance a few weeks ago we had `1149009458` and this week we have `1158009458`
Based on the old algorithm, the last weeks build is later than this weeks build.